jina/serve/executors/__init__.py

```diff
@@ -248,8 +248,8 @@ def workspace(self) -> Optional[str]:
         :return: returns the workspace of the current shard of this Executor.
         """
         workspace = (
-            getattr(self.metas, 'workspace')
-            or getattr(self.runtime_args, 'workspace', None)
+            getattr(self.runtime_args, 'workspace', None)
+            or getattr(self.metas, 'workspace')
             or os.environ.get('JINA_DEFAULT_WORKSPACE_BASE')
         )
```

Comment:
This makes it so that the `workspace` set on the Executor level (in the `exec.yml` or in `metas.workspace`) can be overridden by the `workspace` in the `runtime_args` (so in `.add()`, `flow.yml`, or `docker_compose.yml`), instead of the other way around. This is necessary to be able to put the workspace into the automatically added volume.

In some (I think rare) instances, this can be a **breaking change**. Basically whenever a user set a workspace both on the executor level and the runtime level, the behaviour will change with this.